homeassistant/components/mqtt/__init__.py

```diff
@@ -704,7 +704,6 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
             for component in PLATFORMS
         )
     )
-    await hass.async_block_till_done()
```

Comment:
If you want to be extra safe you could replace it with a `await asyncio.sleep(0)` to yield to the event loop and allow anything to finish

Reply:
Good suggestion I'll do that!